### ISSUE TITLE
Incorrect conversion from sliced pd.MultiIndex

### ISSUE THREAD

**Issue by aseyboldt** [pydata/xarray#2496]
If we convert a pandas dataframe with a multiindex, slice it to remove some entries from the index, a converted DataArray still contains the removed items in the coordinates (although the values are NaN).

```python
# We create an example dataframe
idx = pd.MultiIndex.from_product([list('abc'), list('xyz')])
df = pd.DataFrame(data={'col': np.random.randn(len(idx))}, index=idx)
df.columns.name = 'cols'
df.index.names = ['idx1', 'idx2']
df2 = df.loc[['a', 'b']]
```
```python
# df2 does not contain `c` in the first level
>>> df2
cols            col
idx1 idx2          
a    x    -0.844476
     y    -0.845998
     z     1.965143
b    x    -0.159293
     y     0.188163
     z    -1.076204

# It still shows up in the converted xarray though:
>>> xr.DataArray(df2).unstack('dim_0')
<xarray.DataArray (cols: 1, idx1: 3, idx2: 3)>
array([[[-0.844476, -0.845998,  1.965143],
        [-0.159293,  0.188163, -1.076204],
        [      nan,       nan,       nan]]])
Coordinates:
  * cols     (cols) object 'col'
  * idx1     (idx1) object 'a' 'b' 'c'
  * idx2     (idx2) object 'x' 'y' 'z'
```

If the original dataframe is very sparse, this can lead to gigantic unnecessary memory usage.

<details>

#### Output of ``xr.show_versions()``

```
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.5.final.0
python-bits: 64
OS: Darwin
OS-release: 17.7.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: en_GB.UTF-8
LANG: None
LOCALE: en_GB.UTF-8

xarray: 0.10.9
pandas: 0.23.4
numpy: 1.15.2
scipy: 1.1.0
netCDF4: 1.4.1
h5netcdf: 0.6.2
h5py: 2.8.0
Nio: None
zarr: None
cftime: 1.0.0b1
PseudonetCDF: None
rasterio: None
iris: None
bottleneck: 1.2.1
cyordereddict: None
dask: 0.19.2
distributed: 1.23.2
matplotlib: 3.0.0
cartopy: None
seaborn: 0.9.0
setuptools: 40.4.3
pip: 18.0
conda: 4.5.11
pytest: 3.8.1
IPython: 7.0.1
sphinx: 1.8.1
```

</details>

**Comment by shoyer** [comment]
I suppose we could use [`MultiIndex.remove_unused_levels()`](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.MultiIndex.remove_unused_levels.html#pandas-multiindex-remove-unused-levels) when unstacking -- this appears to be what pandas does:
```
>>> df2.unstack()
cols       col
idx2         x         y         z
idx1
a    -1.470914 -0.093633  1.211639
b     0.587716  1.425005  0.106892
```

I'll mark this as a bug, though it's also a bit of a backwards incompatible change. Any interest in putting together a PR? :)

**Comment by fujiisoup** [comment]
It was fixed by #2619
